File: C++/c-StandardLibraryIncludingC-14-C-17/AdaptorsForContainers/stack.cpp
// stack.cpp
#include <iostream>
#include <stack>

int main(){

  std::stack<int> myStack;

  std::cout << myStack.empty() << std::endl;   // true
  std::cout << myStack.size() << std::endl;    // 0

  myStack.push(1);
  myStack.push(2);
  myStack.push(3);
  std::cout << myStack.top() << std::endl;     // 3

  while (!myStack.empty()){ 
    std::cout << myStack.top() << " ";
    myStack.pop();
  }                                            // 3 2 1

  std::cout << std::endl << myStack.empty() << std::endl;   // 1 (denotes true)
  std::cout << myStack.size() << std::endl;    // 0

  return 0;
}

File: GO/DecodeTheCodingInterviewInGo_RealWorldExamples/Amazon/Feature8MergeRecommendations/Stack.go
package main

type Stack []string

func (s Stack) Push(v string) Stack {
    return append(s, v)
}

func (s Stack) Pop() (Stack, string) {
    l := len(s)
	if l == 0 {
		return s, ""
	}
    return s[:l-1], s[l-1]
}

func (s Stack) Top() string {
    l := len(s)
	return s[l - 1]
}

func (s Stack) Empty() bool {
	return len(s) == 0
}
